\chapter{List of abbreviations}

\noindent \textbf{NHBCS}: New Hampshire Birth Cohort Study \\
\noindent \textbf{PCoA}: Principal Coordinates Analysis \\
\noindent \textbf{NMR}: Nuclear Magnetic Resonance \\
\noindent \textbf{ASV}: Amplicon Sequence Variants \\
\noindent \textbf{OTU}: Operational Taxonomic Unit \\
\noindent \textbf{gUniFrac}: Generalized Unique Fraction \\
\noindent \textbf{sCCA/CCA}: Sparse Canonical Correlation Analysis \\
\noindent \textbf{FDR}: False Discovery Rate \\
\noindent \textbf{RF}: Random Forest \\
\noindent \textbf{ASV}: Amplicon Sequence Variants \\
\noindent \textbf{SCFA}: Short chain fatty acids \\
\noindent \textbf{EN}: Elastic Net \\
\noindent \textbf{SVM-RBF}: Support Vector Machines with Radial Basis Kernel Function \\
\noindent \textbf{SPLS}: Sparse Partial Least Squares \\
\noindent \textbf{CLR}: Centered Log Ratio Transformation \\ 
\noindent \textbf{KEGG}: Kyoto Encyclopedia of Genes and Genomes \\
\noindent \textbf{GOLD}: Genomes OnLine Database \\
\noindent \textbf{CBEA}: Competitive Balances for taxonomic Enrichment Analysis \\
\noindent \textbf{HMDB}: Human Metabolite DataBase \\
\noindent \textbf{HMP}: Human Microbiome Project \\
\noindent \textbf{GSVA}: Gene Set Variation Analysis \\
\noindent \textbf{GSEA}: Gene Set Enrichment Analysis \\
\noindent \textbf{ssGSEA}: Single Sample Gene Set Variation Analysis \\
\noindent \textbf{GO}: Gene Ontology \\
\noindent \textbf{CRC}: Colorectal Cancer \\
\noindent \textbf{IBD}: Inflammatory Bowel Disease \\
\noindent \textbf{MSigDB}: Molecular Signatures Database \\
\noindent \textbf{VAM}: Variance-adjusted Mahalanobis \\
\noindent \textbf{CoDA}: Compositional Data Analysis \\
\noindent \textbf{iNKT}: Invariant natural killer T cells \\
\noindent \textbf{TLR-5}: Toll-like receptor 5 \\
\noindent \textbf{GI}: Gastrointestinal tract \\
\noindent \textbf{DNA}: Deoxyribonucleic acid \\
\noindent \textbf{RNA}: Ribonucleic acid \\
\noindent \textbf{SCC}: Spearman correlation coefficient \\
\noindent \textbf{PICRUSt}: Phylogenetic Investigation of Communities by Reconstruction of Unobserved States \\
\noindent \textbf{ILR}: Isometric log ratio transformation \\
\noindent \textbf{LASSO}: Least absolute shrinkage and selection operator \\
\noindent \textbf{FID}: Free induction decay \\
\noindent \textbf{PMA}: Penalized Multivariate Analysis \\
\noindent \textbf{PCR}: Polymerase Chain Reaction \\
\noindent \textbf{QC}: Quality Control \\
\noindent \textbf{PRESS}: Predicted residual error sum of squares \\
\noindent \textbf{QC}: Quality Control \\
\noindent \textbf{DADA}: Divisive Amplicon Denoising Algorithm \\
\noindent \textbf{AUROC}: Area under the receiver operating characteristic curve \\
\noindent \textbf{QIIME}:  Quantitative Insights Into Microbial Ecology \\
\noindent \textbf{MetaPhlAn}: Metagenomic Phylogenetic Analysis \\
\noindent \textbf{PhILR}: Phylogenetic isometric log ratio transformation \\
\noindent \textbf{SMOTE}: Synthetic Minority Oversampling Technique \\
\noindent \textbf{CDF}: Cumulative distribution function \\
\noindent \textbf{MSEA}: Microbe-set enrichment analysis  \\
\noindent \textbf{NCBI}: National Center of Biotechnology Information \\
\noindent \textbf{LC}: Liquid Chromatography \\
\noindent \textbf{MS}: Mass Spectrometry \\
\noindent \textbf{UBERON}:  Uber-anatomy ontology \\
\noindent \textbf{EFO}: Experimental factor ontology \\
\noindent \textbf{OMP}: Ontology of microbial phenotypes \\
