\chapter{Introduction}

\section{The human gut microbiome and population health}
\subsection{Overview of the human gut microbiome}

The human microbiome is the collection microorganisims (which includes bacteria, protozoa, archaea, fungi, viruses, and their genes) that participate in a symbiotic co-existence with their hosts \cite{ursell2012defining}. It is difficult to study human microbiomes outside the host due to difficulties in culturing the majority of organisms \cite{}, however advances in sequencing technologies have allowed researchers to glimpse into the inner workings of these complex communities \cite{}. Most notably, different body sites harbor unique environmental determinants that give rise to distinct groups of microbes \cite{consortium2012structure}. For example, in the oral microbiome, oral surfaces have different surface receptors \cite{gibbons1989bacterial}, promoting only microbes with specific adhesins that are complementary \cite{aas2005defining}. This resulted in differences such as \emph{Streptococcus mitis} bv.2 species being represted in the tongue dorsum but not even detected in the related lateral tongue surface \cite{aas2005defining}. Even though the degree of diversity across body sites extend across the entire tree of life, most studies so-far have been focused on profiling bacteria, which exists in high-abundance and is relatively easy to profile \cite{}. 

The gut environment also promotes a very specific group of microbes which varies across the digestive track \cite{mailhe2018repertoire, donaldson2016gut}. Most microbiome research has been focused on the colon via fecal samples \cite{tang2020current}, where it has been estimated to contain the highest microbial density recorded in any habitat \cite{}. The gut microbiome is acquired at birth \cite{} via maternal transfer \cite{}. The gut community matures over time, increasing in diversity and reacing an adult-like state is achieved around 2-3 years of age, where it is characterized mainly by members of the Firmicutes, Actinobacteria, and Bacteroidetes phyla, with \emph{Bacteroides}, \emph{Faecalibacterium}, and \emph{Bifidobacterium} as the most abundant genera \cite{king2019baseline, metahitconsortiumadditionalmembers2011enterotypes}. Many of the species identified to be in the gut microbiome cannot be found in other habitats, suggesting a strong co-evolutionary relationship with human hosts \cite{ley2006ecological}.  

Various environmental factors can shape the composition of the gut microbiome. In early life, the mode of delivery and breastfeeding status are significant modifiers of composition. Infants who were born via vaginal delivery has increased abundances of \emph{Bacteroides}, \emph{Pectobacterium}, and \emph{Bifidobacterium} genera, while those born via Cesarian section have decreased diversity and higher propensity to be colonized by \emph{Staphylococcus} and members of the \emph{Clostridum} cluster \cite{madan2016effects, kim2020delayed, stewart2018temporal}. Breastfeeding is assocated with lower levels of \emph{Escherichia coli}, \emph{Tyzzerella nexilis}, and \emph{Roseburia intestinalis} while on the other hand promoting the coloization of various \emph{Bifidobacterium} species such as \emph{B. breve}, \emph{B. dentium} which harbors specific genes that help in digesting complex oligosaccharides \cite{stewart2018temporal, vatanen2018human}. Among adults, diet is of particular interest. Studies have shown that the Western diet, high in saturated and trans fats while low in mono and polyunsaturated fats, is associated with decreased abundance in \emph{Bifidobacterium}, \emph{Eubacterium}, and \emph{Lactobacillus} genera \cite{wu2011linking}. Other extrinsic and intrinsic factors such as smoking status \cite{biedermann2013smoking}, alcohol consumption \cite{dubinkina2017links}, etc. also contribute to microbiome modulation (reviewed in \cite{schmidt2018humana}). In conclusion, the gut microbiome is highly malleable yet intrinsic part of human existence. 

\subsection{Outcomes associated with changes in gut microbiome composition}

Observed shifts in gut microbiome composition are often associated with adverse health outcomes. As such, there is a great interest in epidemiological applications, where the microbiome can be identified as either a marker or as a causal agent in human disease \cite{foxman2015use}. Observational studies have linked changes in the gut microflora to various metabolic and infectious diseases. This is because host inflammation responses can be linked to the microbiome's role in mediating immune function (reviewed in \cite{wang2020relationship}). Short-chain fatty acids (SCFAs), such as acetate, propionate, and butyrate, are metabolic products of microbiota digestion from dietary fiber and resistant starches. These compounds bind to G-protein-coupled receptor 43 expressed in immune cells, an effect that allows for resolution of immune responses and prevent inflammation. As such, the gut microbiome is found to be associated with inflammation-related diseases such as colorectal cancer \cite{cheng2020intestinal, yu2017metagenomica} and inflammatory bowel disease \cite{gevers2014treatmentnaive, franzosa2019gut, lloyd-price2019multiomics}. There are many other conditions (reviewed in \cite{cho2012human}) that can be linked to changes in the gut microbiome, such as \emph{Clostridium difficile} infection \cite{weingarden2014microbiota} and obesity \cite{turnbaugh2009core, aoun2020influence}.   

Additionaly, the gut microbiome is also linked to other conditions that is not localized in the intestinal track. The gut-brain axis refers to how residential gut microbes are also involved in regulating host behaviour \cite{morais2021gut}. This has linked the gut microbiome to neurological conditions such as autisum spectrum disorder (ASD), where patients with ASD have different gut microbial signatures (an increase in several mucosa-associated Clostridiales) compared to neurotypical controls \cite{luna2017distinct}. Another instance of the gut microbiome's overarching impact on human physiology is the gut-lung axis \cite{enaud2020gutlung}, where the crosstalk between gut and lung communities is linked to chronic and acute respiratory conditions. For example, a study from the Canadian Healthy Infant Longitudinal Development cohort identified decreases in \emph{Lachnospira}, \emph{Veillonella}, \emph{Faecalibacterium}, and \emph{Rothia} genera among children at risk of asthma attacks \cite{arrietta2015early}. 


\subsection{Challenges in determining potential microbial biomarkers}
Despite the wealth of microbiome association studies, there still exists considerable challenges in identifying consistent microbial markers that have meaningful associations with health outcomes \cite{duvallet2017metaanalysis}. In a meta-analysis of 10 studies for inflammatory bowel disease and obesity \cite{walters2014metaanalyses}, no individual microbe was consistently associated with subject case status. Even though a degree of false positive is expected since population-level studies are indeed only exploratory and hypothesis generating, the practice of interpreting differentially abundant taxa lists using post-hoc literature searches makes results unreliable and biased. As a result, validating these markers proves to be arduous as it is impossible to disentangle which hypotheses are more contextually probable to follow-up in expensive \emph{in vitro} or \emph{in vivo} laboratory experiments.  

Statistical and computational difficulties remain one of the major hurdles towards this replication issue \cite{li2015microbiome, li2019comparative}. Since most microbes cannot be cultured, researchers often rely on high-throughput sequencing technologies to profile the abundance of each microbes such as 16S rRNA gene amplicon sequencing. Various steps within the sample processing protocol such as storage, DNA extraction, and library preparation, can contribute to differences in results between studies \cite{clausen2021evaluating}. Specifically, contamination remains a big issue \cite{}, especially in environments where there is low total microbial load \cite{}, thereby producing erroneous results \cite{}. Additionally, bioinformatic analyses and usage of different databases might also cause divergences in observations \cite{moossavi2020biological}. For example, choosing to cluster amplicon sequence variants (ASVs) to operational taxonomic units (OTUs) using a sequence similarity threshold can yield radically different communities \cite{chiarello2022ranking, moossavi2020biological}.   

In addition to issues pertaining to converting raw sequence data to named microbes, statistical analyses used to identify relevant candidates also suffers from inconsistencies. Microbiome taxonomic data, like other sequencing data sets, is compositional \cite{gloor2017microbiome, quinn2019field} and constrained by the total number of reads (or library size). This constraint induces spurious negative correlations between variables, whereby changes in true absolute counts might not be accurately reflected at the observed relative scale \cite{lin2020analysis, morton2019establishing}. Microbiome data is sparse, containing a mixture of both structural zeroes (true absence of a taxon), and technical zeroes (abundance below the limit of detection) \cite{kaul2017analysis, silverman2020naught}. This makes it difficult to distinguish between low-abundance and absent taxa, especially when studies differ considerably in sequencing quality and depth. Finally, microbiome data is also high-dimensional, where a typical data set contains from hundreds of species to thousands of sequence variants resulting in high multiple testing burden. All of the challenges above contribute to methods that have to make difficult trade-offs between power, type I error control, and effect size estimations. As a consequence, researchers are faced with a dizzyingly complex landscape of available methodology, which has been shown to produce different results performed on the same data set \cite{nearing2022microbiome}.   

In addition to technological challenges, the gut microbiome itself is also dynamic with a great degree of intra-individual variability. Even though the adult gut microbiome composition is stable over longer time scales \cite{consortium2012structure}, studies have also shown that shifts in composition occur on a day-to-day basis, impacted by host diet and lifestyle \cite{david2014host, david2014diet}. However, the magnitude of shifts are small when compared to between host variability. In the original human microbiome project (HMP) paper, researchers estimated that for habitat-specific signature genera, their presence in corresponding sampling sites is between 17\% and 84\% \cite{consortium2012structure}. In another study of gut microbiomes from a Chinese cohort of healthy individuals (N = 120), around 90 ($\pm 16$) species-level taxa (assayed using full length 16S rRNA gene sequencing) were shared between individuals from the total of 1,235 identified species \cite{yang2020species}. This effect is consistent even when looking at the strain level \cite{lloyd-price2017strains}. For context, humans share around 99.5\% of their genomes \cite{genetics}. The degree of microbiome personalization is significant enough to warrant initial exploration of forensic applications \cite{fierer2010forensic}. 


\section{Approaching microbiome research from a mechanistic perspective}

\subsection{The functional microbiome}

The end-goal of medical and epidemiological research on the gut microbiome is to ascertain why selected microbes exist in a given environment, and how they can affect human physiology to cause or mediate disease. The questions of ``why" and ``how" usually underline discussions surrounding microbial function \cite{klassen2018defining}, the answers to which can allow for the design of therapeutics and interventions \cite{durack2019gut}. As such, researchers are interested in moving beyond looking at the microbiome from a purely taxonomic perspective to a functional one \cite{heinz-buschart2018human}.  

One of the most important and well known roles of the gut microbiome involves the fermentation of foods into metabolites that can be absorbed by its host. Studies have shown that the microbiome is involved in the digestion of all three macronutrient sources from the human diet: carbohydrates, lipids, and proteins (reviewed in \cite{oliphant2019macronutrient}). For carbohydrates, even though the human gut can hydrolyze and absorb certain sugars such as glucose, fructose, and lactose in the proxmial gastrointestinal (GI) tract, the complexity of bonds that exist between monosaccharide units in dietary starches (especially plant polysaccharides) means that the majority of carbohydrates pass through to the distal GI where it is digested by very well-equipped microbes \cite{wong2006colonic}. The species \emph{Bacteroides thetaiotaomicron} alone has 260 glycoside hydrolases in its genome \cite{xu2003genomic}, well beyond that of native human enzymes. The by-product of carbohydrate fermentation are SFCAs (most notably acetate, proprionate, and butyrate) \cite{macfarlane2012bacteria}, which are compounds that not just have inherent nutritional value but also plays a role in maintaining the gut epithelial barrier. Even though proteins and fats are not as core to microbiome function as that of carbohydrates, certain important compounds have been shown to be linked to gut communities (reviewed in \cite{morowitz2011contributions}). In a study by Backhed et al., germ-free mice were fed significantly more chow compared to wild-type controls yet had 42\% less body fat \cite{backhed2004gut}. Surprisingly, when colonized with even a single species from wild-type microbiomes (also \emph{Bacteroides thetaiotaomicron}), these mice were able to partially recover their capacity to produce body fat, thereby suggesting a relationship between gut microbes and the process of adiposity. This provides further evidence for a link between the gut microbiome and obesity \cite{turnbaugh2009core}. 

Additionally, the microbiome is also implicated with immune programming. Even though the specific signalling pathways are still under active research, this process is mediated through commensal secreted metabolites or surface-associated antigens that interact directly with host immune cells (reviewed in \cite{belkaid2014role}). For example, mice studies have shown that the innate immune receptor Toll-like receptor 5 (TLR5) selects for certain microbes during the neonatal period by serving as a sensor for bacterial flagellin \cite{fulde2018neonatal}. During early life, in order to accommodate the initial colonization process, host immune response is limited and certain types of cell activity is suppressed. Microbes can directly induce this by secreting molecules such as sphingolipids that inhibit the induction of invariant natural killer T cells (iNKT) \cite{an2014sphingolipids}. Even though this process results in an increased propensity of being infected, it also suppresses the host's inflammatory response, preventing excessive behaviours that might cause harmful outcomes such as necrotizing enterocolitis \cite{neu2011necrotizing}. This crosstalk continues throughout life, where the microbiome participates in a co-operative relationship that helps maintain intestinal homeostatsis (reviewed in \cite{zheng2020interactions}). 

At the end of the day, the list of possible biochemical pathways in which the microbiome participates in is endless, with new studies exploring new and exciting avenues and mechanisms. However, it is clear that profile microbial function can help get at the questions of ``why" certain microbes exist, and ``how" they can lead to positive or negative health outcomes.  

\subsection{Approaches to characterize function in epidemiological studies}

Various advancements in high-throughput molecular technologies have allowed researchers to comprehensively profile different functional components of the microbiome \cite{foxman2015use}. Even though they are powerful, each meta-omic method face different challenges when attempting to measure up to their intended roles. Here, we give a short review of each technology and what type of insight they can give.  

\subsubsection{Metagenomics}
Metagenomics refers to untargeted DNA sequencing of the entire gene content of a sample via ``shotgun" shearing of fragments \cite{quince2017shotgun}. In terms of taxonomic profiling, metagenomic approaches can provide species to strain level resolution \cite{truong2015metaphlan2}, as well as being able to detect non-bacterial organisms such as archaea and viruses. In terms of function, researchers can estimate the abundance of certain gene families in the entire community, which can be used downstream to infer entire pathways and even predict structural variants \cite{kiefl2022structureinformed}. Finally, assembly-based approaches can categorize sequence fragments into putative genomes, thereby enabling the discovery of novel strains and genes \cite{perez-cobas2020metagenomic}. However, since this is a DNA-based approach, gene family copy numbers only represent functional potential rather than true outputs. This is further complicated by the fact that it is challenging to discern which bacteria are ``alive" since DNA is a stable molecule \cite{quince2017shotgun}. As such, inferences about microbiome function drawn from metaegenomic data sets are limited, and difficult to trace back to specific microbes. Finally, databases are woefully biased towards easily identifiable strains, whereby most genes are defined as unmapped, making annotating specific functions difficult. 

\subsubsection{Metatranscriptomics and Metaproteomics}
Metatranscriptomics and metaproteopmics refer to profiling the entire transcript (i.e. RNA) and protein content of a sample \cite{franzosa2015sequencing}. Both techniques look at downstream products of gene abundances, therefore they are better representations the total amount of functional information that is ``active" within a community. Additionally, both metatranscriptomics and metaproteomics can be mapped back to sequences obtained from original metagenomics results (in fact, with proper tagging, metatranscriptomics can be sequenced simultaneously). This allows for powerful multi-omics approaches that can identify how a function resevoir is activated and expressed downstream. For example, in a paired metagenomic-metatranscriptomic analysis of healthy gut microbiomes, researchers found that even though there is a resevoir of genes coding for biosynthesis of amino acids, low transcriptional activity suggests that this function is under-expressed \cite{franzosa2014relating}. This is consistent with the fact that these procedures are energetically unfavorable \cite{oliphant2019macronutrient}. Despite such benefits, there are various challenges in the technical aspect of these profiling approaches. For metatranscriptomics, not only are mRNAs highly unstable, they are dwarfed in abundance by rRNA in the total RNA pool, requiring specialized techniques to enrich for them while also removing human contaminants. For metaproteomics, 

\subsubsection{Metabolomics}



 

\subsection{Obstacles in a function-based approach} 

Microbiome functional profiles are relatively more stable across individuals \cite{consortium2012structure}. The degree of comparative stability is difficult to ascertain due to differences in the scale of comparison across taxonomy and function \cite{langille2018exploring}. 

\section{An integrative approach to incorporate both structure and function}

\subsection{Importance of taxa-function relationships}
Here we talk about the definition of the taxa-function relationship and how they are important. The key idea here is that therapeutics and environmental factors can influence taxa but then those taxa interact with other members of the community to form functional components that interact with the host. 


\subsection{Biodiversity, ecosystem functioning, and community resilience}
Taxa-function relationships are part of an ecology concept titled "biodiversity and ecosystem functioning" which relates to community resilience and probably long-term health outcomes. 


\section{Challenges and opportunities in deciphering the taxa-function relationship}
