\chapter{Introduction}

\section{The human gut microbiome and population health}
\subsection{Overview of the human gut microbiome}

The human microbiome is the collection of microorganisims (which includes bacteria, protozoa, archaea, fungi, viruses, and their genes) that participate in a symbiotic co-existence with their hosts \cite{ursell2012defining}. It is difficult to study human microbiomes outside the host due to difficulties in culturing the majority of organisms \cite{}, however, advances in sequencing technologies have allowed researchers to glimpse the inner workings of these complex communities \cite{}. Most notably, different body sites harbor unique environmental determinants that give rise to distinct groups of microbes \cite{consortium2012structure}. For example, in the oral microbiome, oral surfaces have different surface receptors \cite{gibbons1989bacterial}, promoting only microbes with specific adhesins that are complementary \cite{aas2005defining}. This results in differences such as \emph{Streptococcus mitis} bv.2 species being represented in the tongue dorsum but not even detected in the related lateral tongue surface \cite{aas2005defining}. Even though the degree of diversity across body sites extend across the entire tree of life, most studies so-far have focused on profiling bacteria, which exists in high-abundance and is relatively easy to profile \cite{}. 

The gut environment also promotes a very specific group of microbes which varies across the digestive track \cite{mailhe2018repertoire, donaldson2016gut}. Most microbiome research has been focused on the colon via fecal samples \cite{tang2020current}, where it has been estimated to contain the highest microbial density recorded in any habitat \cite{}. The gut microbiome is acquired at birth \cite{} via maternal transfer \cite{}. The gut community matures over time, increasing in diversity and reaching an adult-like state in around 2-3 years of age, where it is characterized mainly by members of the Firmicutes, Actinobacteria, and Bacteroidetes phyla, with \emph{Bacteroides}, \emph{Faecalibacterium}, and \emph{Bifidobacterium} as the most abundant genera \cite{king2019baseline, metahitconsortiumadditionalmembers2011enterotypes}. Many of the species identified to be in the gut microbiome cannot be found in other habitats, suggesting a strong co-evolutionary relationship with human hosts \cite{ley2006ecological}.  

Various environmental factors can shape the composition of the gut microbiome. In early life, the mode of delivery and breastfeeding status are significant modifiers of composition. Infants who were born via vaginal delivery have increased abundances of \emph{Bacteroides}, \emph{Pectobacterium}, and \emph{Bifidobacterium} genera, while those born via Cesarian section have decreased diversity and higher propensity to be colonized by \emph{Staphylococcus} and members of the \emph{Clostridum} cluster \cite{madan2016effects, kim2020delayed, stewart2018temporal}. Breastfeeding is associated with lower levels of \emph{Escherichia coli}, \emph{Tyzzerella nexilis}, and \emph{Roseburia intestinalis} while on the other hand promoting the coloization of various \emph{Bifidobacterium} species such as \emph{B. breve} and \emph{B. dentium} which harborsspecific genes that help in digesting complex oligosaccharides \cite{stewart2018temporal, vatanen2018human}. Among adults, diet is of particular interest. Studies have shown that the Western diet, high in saturated and trans fats while low in mono and polyunsaturated fats, is associated with decreased abundance in \emph{Bifidobacterium}, \emph{Eubacterium}, and \emph{Lactobacillus} genera \cite{wu2011linking}. Other extrinsic and intrinsic factors such as smoking status \cite{biedermann2013smoking} and alcohol consumption \cite{dubinkina2017links} also contribute to microbiome modulation (reviewed in \cite{schmidt2018humana}). In conclusion, the gut microbiome is highly malleable yet intrinsic part of human existence. 

\subsection{Outcomes associated with changes in gut microbiome composition}

Observed shifts in gut microbiome composition are often associated with adverse health outcomes. As such, there is a great interest in epidemiological applications, where the microbiome can be identified as either a marker or as a causal agent in human disease \cite{foxman2015use}. Observational studies have linked changes in the gut microflora to various metabolic and infectious diseases. This is because host inflammation responses can be linked to the microbiome's role in mediating immune function (reviewed in \cite{wang2020relationship}). Short-chain fatty acids (SCFAs), such as acetate, propionate, and butyrate, are metabolic products of microbiota digestion from dietary fiber and resistant starches. These compounds bind to G-protein-coupled receptor 43 expressed in immune cells, an effect that allows for resolution of immune responses and prevention of inflammation. As such, the gut microbiome is found to be associated with inflammation-related diseases such as colorectal cancer \cite{cheng2020intestinal, yu2017metagenomica} and inflammatory bowel disease \cite{gevers2014treatmentnaive, franzosa2019gut, lloyd-price2019multiomics}. There are many other conditions (reviewed in \cite{cho2012human}) that can be linked to changes in the gut microbiome, such as \emph{Clostridium difficile} infection \cite{weingarden2014microbiota} and obesity \cite{turnbaugh2009core, aoun2020influence}.   

Additionally, the gut microbiome is also linked to other conditions that are not localized in the intestinal tract. The gut-brain axis refers to how residential gut microbes are involved in regulating host behavior \cite{morais2021gut}. There are established links, for example, between the gut microbiome and neurological conditions such as autisum spectrum disorder (ASD), where individuals with ASD have different gut microbial signatures (an increase in several mucosa-associated Clostridiales) compared to neurotypical controls \cite{luna2017distinct}. Another instance of the gut microbiome's overarching impact on human physiology is the gut-lung axis \cite{enaud2020gutlung}, where the crosstalk between gut and lung communities is linked to chronic and acute respiratory conditions. For example, a study from the Canadian Healthy Infant Longitudinal Development cohort identified decreases in \emph{Lachnospira}, \emph{Veillonella}, \emph{Faecalibacterium}, and \emph{Rothia} genera among children at risk of asthma attacks \cite{arrieta2015early}. 


\subsection{Challenges in determining potential microbial biomarkers}
Despite the wealth of microbiome association studies, there still exist considerable challenges in identifying consistent microbial markers that have meaningful associations with health outcomes \cite{duvallet2017metaanalysis}. In a meta-analysis of 10 studies for inflammatory bowel disease and obesity \cite{walters2014metaanalyses}, no individual microbe was consistently associated with subject case status. Even though false positives are expected since population-level studies are indeed only exploratory and hypothesis generating, the practice of interpreting differentially abundant taxa lists using post-hoc literature searches makes results unreliable and biased. As a result, validating these markers proves to be arduous as it is impossible to disentangle which hypotheses are more contextually probable to follow-up in expensive \emph{in vitro} or \emph{in vivo} laboratory experiments.  

Statistical and computational difficulties remain one of the major hurdles towards this replication issue \cite{li2015microbiome, li2019comparative}. A mainstay of research in the field is high-throughput sequencing technologies to profile microbial communities taxonomically and genomically. Various steps within the sample processing protocol such as storage, DNA extraction, and library preparation, can contribute to differences in results between studies \cite{clausen2021evaluating}. Specifically, contamination remains a big issue \cite{}, especially in environments where there is low total microbial load \cite{}, thereby producing erroneous results \cite{}. Additionally, bioinformatic analyses and use of different databases may also cause divergences in observations \cite{moossavi2020biological}. For example, when 16S rRNA gene amplicon sequencing is used for taxonomic profiling, choosing to cluster amplicon sequence variants (ASVs) to operational taxonomic units (OTUs) using a sequence similarity threshold can yield radically different communities \cite{chiarello2022ranking, moossavi2020biological}.   

In addition to issues pertaining to converting raw sequence data to named microbes, statistical analyses used to identify relevant candidates also suffer from inconsistencies. Microbiome taxonomic data, like other sequencing data sets, is compositional \cite{gloor2017microbiome, quinn2019field} and constrained by the total number of reads (or library size). This constraint induces spurious negative correlations between variables, whereby changes in true absolute counts might not be accurately reflected at the observed relative scale \cite{lin2020analysis, morton2019establishing}. Microbiome data is sparse, containing a mixture of both structural zeroes (true absence of a taxon), and technical zeroes (abundance below the limit of detection) \cite{kaul2017analysis, silverman2020naught}. This makes it difficult to distinguish between low-abundance and absent taxa, especially when studies differ considerably in sequencing quality and depth. Finally, microbiome data is also high-dimensional, where a typical data set contains from hundreds of species to thousands of sequence variants resulting in a high multiple testing burden. All of the challenges above contribute to methods that have to make difficult trade-offs between power, type I error control, and effect size estimations. As a consequence, researchers are faced with a complex landscape of available methods, which have been shown to produce different results on the same data sets \cite{nearing2022microbiome}.   

In addition to technological challenges, the gut microbiome itself is also dynamic with a great degree of intra-individual variability. Even though the adult gut microbiome composition is stable over longer time scales \cite{consortium2012structure}, studies have also shown that shifts in composition occur on a day-to-day basis, impacted by host diet and lifestyle \cite{david2014host, david2014diet}. However, the magnitude of shifts are small when compared to between host variability. In the original human microbiome project (HMP) paper, researchers estimated that for habitat-specific signature genera, their presence in corresponding sampling sites is between 17\% and 84\% \cite{consortium2012structure}. In another study of gut microbiomes from a Chinese cohort of healthy individuals (N = 120), around 90 ($\pm 16$) species-level taxa (assayed using full length 16S rRNA gene sequencing) were shared between individuals from the total of 1,235 identified species \cite{yang2020specieslevel}. This effect is consistent even when looking at the strain level \cite{lloyd-price2017strains}. For context, humans share around 99.5\% of their genomes \cite{genetics}. The degree of microbiome personalization is significant enough to warrant initial exploration of forensic applications \cite{fierer2010forensic}. 


\section{Approaching microbiome research from a mechanistic perspective}

\subsection{The functional microbiome}

The end-goal of medical and epidemiological research on the gut microbiome is to ascertain why selected microbes exist in a given environment, and how they can affect human physiology to cause or mediate disease. The questions of ``why" and ``how" usually underline discussions surrounding microbial function \cite{klassen2018defining}, the answers to which can allow for the design of therapeutics and interventions \cite{durack2019gut}. As such, researchers are interested in moving beyond looking at the microbiome from a purely taxonomic perspective to a functional one \cite{heintz-buschart2018human}.  

One of the most important and well-known roles of the gut microbiome involves the fermentation of foods into metabolites that can be absorbed by its host. Studies have shown that the microbiome is involved in the digestion of all three macronutrient sources from the human diet: carbohydrates, lipids, and proteins (reviewed in \cite{oliphant2019macronutrient}). For carbohydrates, even though the human gut can hydrolyze and absorb certain sugars such as glucose, fructose, and lactose in the proximal gastrointestinal (GI) tract, the complexity of bonds that exist between monosaccharide units in dietary starches (especially plant polysaccharides) means that the majority of carbohydrates pass through to the distal GI where it is digested by very well-equipped microbes \cite{wong2006colonic}. The species \emph{Bacteroides thetaiotaomicron} alone has 260 glycoside hydrolases in its genome \cite{xu2003genomic}, well beyond that of native human enzymes. The by-product of carbohydrate fermentation are SFCAs (most notably acetate, proprionate, and butyrate) \cite{macfarlane2012bacteria}, which are compounds that not only have inherent nutritional value but also play a role in maintaining the gut epithelial barrier. Even though proteins and fats are not as central to microbiome function as carbohydrates, certain important compounds have been shown to be linked to gut communities (reviewed in \cite{morowitz2011contributions}). In a study by Backhed et al., germ-free mice were fed significantly more chow compared to wild-type controls yet had 42\% less body fat \cite{backhed2004gut}. Surprisingly, when colonized with even a single species from wild-type microbiomes (also \emph{Bacteroides thetaiotaomicron}), these mice were able to partially recover their capacity to produce body fat, thereby suggesting a relationship between gut microbes and the process of adiposity. This provides further evidence for a link between the gut microbiome and obesity \cite{turnbaugh2009core}. 

The microbiome is also implicated with immune programming. Even though the specific signalling pathways are still under active research, this process is mediated through commensal secreted metabolites or surface-associated antigens that interact directly with host immune cells (reviewed in \cite{belkaid2014role}). For example, mice studies have shown that the innate immune receptor Toll-like receptor 5 (TLR5) selects for certain microbes during the neonatal period by serving as a sensor for bacterial flagellin \cite{fulde2018neonatal}. During early life, in order to accommodate the initial colonization process, host immune response is limited and certain types of cell activity is suppressed. Microbes can directly induce this by secreting molecules such as sphingolipids that inhibit the induction of invariant natural killer T cells (iNKT) \cite{an2014sphingolipids}. Even though this process results in an increased propensity of being infected, it also suppresses the host's inflammatory response, preventing excessive behaviors that might cause harmful outcomes such as necrotizing enterocolitis \cite{neu2011necrotizing}. This crosstalk continues throughout life, where the microbiome participates in a co-operative relationship that helps maintain intestinal homeostatsis (reviewed in \cite{zheng2020interaction}). 

Overall, the list of possible biochemical pathways in which the microbiome participates is endless, with new studies exploring exciting avenues and mechanisms. However, it is clear that profiling microbial function can help get at the questions of ``why" certain microbes exist, and ``how" they can lead to positive or negative health outcomes.  

\subsection{Approaches to characterize function in epidemiological studies}

Advances in high-throughput molecular technologies have allowed researchers to comprehensively profile different functional components of the microbiome \cite{foxman2015use}. Although powerful, each meta`omic method faces different challenges when attempting to measure up to their intended roles. Here, we give a short description of major microbiome profiling technologies currently in use and the types of biological insight they can provide.  

\subsubsection{Metagenomics}
Metagenomics refers to untargeted DNA sequencing of the entire gene content of a sample via ``shotgun" shearing of fragments \cite{quince2017shotgun}. In terms of taxonomic profiling, metagenomic approaches can provide species to strain level resolution \cite{truong2015metaphlan2}, as well as being able to detect non-bacterial organisms such as archaea and viruses. In terms of function, researchers can estimate the abundance of certain gene families in the entire community, which can be used downstream to infer entire pathways and even predict structural variants \cite{kiefl2022structureinformed}. Finally, assembly-based approaches can categorize sequence fragments into putative genomes, thereby enabling the discovery of novel strains and genes \cite{perez-cobas2020metagenomic}. However, since this is a DNA-based approach, gene family copy numbers only represent functional potential rather than true outputs. This is further complicated by the fact that it is challenging to discern which bacteria are ``alive" since DNA is a stable molecule \cite{quince2017shotgun}. As such, inferences about microbiome function drawn from metaegenomic data sets are limited, and difficult to trace back to specific microbes. Finally, databases are woefully biased towards easily identifiable strains, whereby most genes are defined as unmapped, making annotating specific functions difficult. 

\subsubsection{Metatranscriptomics and Metaproteomics}

Metatranscriptomics and metaproteopmics refer to profiling the entire transcript (i.e. RNA) and protein content of a sample \cite{franzosa2015sequencing}. Both techniques measure downstream products of gene abundances, therefore they are better representations of the total amount of functional information that is ``active" within a community. Additionally, both metatranscriptomics and metaproteomics can be mapped back to sequences obtained from original metagenomics results (in fact, with proper tagging, metatranscriptomics can be simultaneously sequenced). This allows for powerful multi-omics approaches that can identify how a function reservoir is activated and expressed downstream. For example, in a paired metagenomic-metatranscriptomic analysis of healthy gut microbiomes, researchers found that even though there is a reservoir of genes coding for biosynthesis of amino acids, low transcriptional activity suggests that this function is under-expressed \cite{franzosa2014relating}. This is consistent with the fact that these procedures are energetically unfavorable \cite{oliphant2019macronutrient}. Despite such benefits, there are various challenges in the technical aspect of these profiling approaches. For metatranscriptomics, not only are mRNAs highly unstable, they are dwarfed in abundance by rRNA in the total RNA pool, requiring specialized techniques to enrich for them while also removing human contaminants. For metaproteomics, the process of protein purification and extraction is demanding due to the complexity of the environment, requiring more biomass as well as special sample preparations to reach the degree of depth obtained by nucleic acid sequencing technologies \cite{leary2013which, schiebenhoefer2019challenges, verberkmoes2009shotgun}. 

\subsubsection{Metabolomics}
Metabolomics refers to the direct quantification of metabolites and other small molecules. It is different from  metatranscriptomics and metaproteomics in the fact that there is no convenient map sequence information \cite{franzosa2015sequencing}, making direct integration with taxonomic or gene abundances from metagenomic sequencing more challenging. However, metabolomics reflects the layer of microbiome function that is closest to the host-microbiome interface, as measured metabolites interact directly with host receptors or participate in collaborative metabolic pathways \cite{tang2011microbial}. Integrative multi-omics data sets featuring metabolomics have shown important links between microbes, their metabolic outputs, and human disease. For example, a study identified that changes in the microbiome have been implicated in the production of trimethylamine N-oxide (TMAO) from l-carnitine (which is a compound that is commonly found in red meat) \cite{wang2011gut}. One large benefit of metabolomics is that its sample preparation requirements are not as demanding as that of metaproteomics \cite{franzosa2019gut, verberkmoes2009shotgun}. However, each molecule has different properties and chemical structures, which means that some compounds are easier to measure than others, creating biases in which type of features gets measured \cite{tang2011microbial}. Finally, metabolomic profiles are highly variable and sensitive to perturbations such as food consumption prior to measurement \cite{hollywood2006metabolomics}. As such, it is suggested that metabolite flux might be a more meaningful measure of microbiome function rather than cross-sectional measures of concentration.  


\subsection{Obstacles in a function-based approach} 

A common misconception is that microbiome functional profiles are relatively more stable across individuals compared to their taxonomic counterparts \cite{consortium2012structure}. This means that analyses focusing on community-wide functional `omics analyses might produce more consistent and validatable results. However, the degree of comparative stability is difficult to ascertain due to differences in the scale of comparison across taxonomy and function \cite{langille2018exploring}. In fact, when gene families are considered instead of pathways, the degree of stability decreased significantly \cite{inkpen2017coupling}. Additionally, empirical studies have also shown that using pathway abundances are not significantly better at classifying patients disease status \cite{xu2014which}.   

Two major challenges exist for function-driven microbiome analyses (reviewed in \cite{heintz-buschart2018human}). First, the large number of available molecular technologies mean that there is considerable choice in what constitutes as ``function" in a certain context. In other words, depending on the research question, researchers have to make a decision on the analytical unit of microbial function, be it gene family abundance, pathway presence-absence, or concentrations of groups of metabolites. For example, in an analysis of strain-specific functional adaptation of the infant gut microbiome from the DIABIMMUNE cohort \cite{vatanen2018human}, the authors were interested in microbial capacity to digest human milk oligosaccharides (HMO) as a core ecosystem function. However, HMO metabolism is not encoded as a single pathway in frequently used databases such as the Kyoto Encyclopedia of Genes and Genomes (KEGG) or Gene Ontology (GO), but rather as a group of 30 genes identified via literature review. As such, defining functions based on the research question of interest can provide meaningful interpretations on the services that microbes confer. Second, there is a considerable amount of ``microbial dark matter" that hampers the characterization of functional dynamics \cite{jiao2021microbial}. For example, in a study involving metatranscriptomic profiling, around 9\% of differentially abundant transcripts had unknown function \cite{heintz-buschart2017integrated}. Interpretations without acknowledging the current limitations of our annotation pipeline might lead to misleading results. 

\section{An integrative approach to incorporate both structure and function}

\subsection{Importance of taxa-function relationships}

A holistic understanding of microbiome-related processes requires a conception of the relationship between taxonomic compositions and their functional profiles, termed the taxa-function relationship \cite{langille2018exploring, heintz-buschart2018human}. Unfortunately, limited number of studies have explored taxonomic drivers of functional shifts beyond anecdotal searches in the literature \cite{manor2017systematic}, where it is more common to study them independently. This gap is problematic because even though drivers of the microbiome's impact on host health are its functional outputs, modulation can only occur at the taxonomic level. Niche differentiation driven by abiotic factors, such as the availability of nutrients, shape community assembly \cite{pereira2017microbial}. As such, any attempt to design restorative interventions or understand environmental perturbations requires the ability to pinpoint exact groups of taxon relevant to the functional processes of interest \cite{wong2019new}.   

The taxa-function relationship is also integral to the complex ecological processes that exist within the microbiome. The plasticity of certain functions to external perturbations can be attributed to redundancies in the number of contributing strains \cite{walker1992biodiversity, moya2016functional}. This idea of ``robustness" \cite{eng2018taxafunction} can be used as a proxy to diagnose community-wide or function-specific dysbiosis \cite{vieira-silva2016species}. This also helps explain the underlying mechanisms as to why healthy microbiomes are found to be generally more ``diverse", fitting with the prevailing ecological theory linking biodiversity and ecosystem functioning in microbial systems \cite{tillman2014biodiversity}. On the other hand, if a core function is only contributed by a single taxon, targeted therapies can be designed to improve growth conditions or to provide the necessary probiotic supplements.  

Finally, taxa-function relationships can also enhance quality of existing population-level studies. Providing the relevant context that can increase interpretability, while also help researchers distinguish possible false positives when considering targets for validation. 

\subsection{Challenges and opportunities}

Many studies have attempted to systematically tackle taxon-function integration in microbiome studies \cite{manor2017systematic, vieira-silva2016species, eng2018taxafunction, noecker2019defining}. However, they face challenges regarding simplistic assumptions, limitations in resolution, and biases in reference databases. For example, FishTaco \cite{manor2017systematic}, a tool to estimate species contributions to functional shifts, assumes that the functional content of contributing microbes is consistent at the species level, ignoring strain-level variation. Other ecological processes such as inter-taxa interactions and horizontal gene transfer can also radically change the taxa-function landscape, whereby removals or additions of certain contributing species might affect other seemingly unrelated members. Additionally, there are difficulties in defining relevant functions. Even though pathway annotations from reference databases such as KEGG are informative, it is still difficult to interpret long lists of pathways identified as differentially abundant. 

However, despite such drawbacks, there are numerous opportunities. Utilizing different multi-omic approaches can provide additional insight into ``active" functions whose products are persistent in the gut environment \cite{jiang2016metatranscriptomic}. This is particularly important considering that microbes might harbor genes but can choose not to express them depending on environmental or ecological conditions \cite{metagenomicsofthehumanintestinaltractmetahitconsortium2016transcriptional}. Defining functions in terms of ecosystem roles such as traits \cite{weissman2021exploring} can allow for more interpretable results that are relevant to the condition of interest. This is can allow for context specific functions such as in the aforementioned study by Vatanen et al. \cite{vatanen2018human} where multiple related gene clusters are simultaneously evaluated.   